src/jit/flowgraph.cpp

```diff
@@ -5763,7 +5772,7 @@ void Compiler::fgFindBasicBlocks()
             impInlineInfo->iciCall->gtCallMoreFlags |= GTF_CALL_M_DOES_NOT_RETURN;
```

Comment:
This flag setting is redundant now, isn't it? The assert above could now check either there are no return blocks and the flag was set, OR, there are return blocks and the flag is not set.